mobile-app/app/screens/AppNavigator/screens/Loans/screens/AddOrRemoveCollateralScreen.tsx

```diff
@@ -361,54 +361,81 @@ export function AddOrRemoveCollateralScreen({ route }: Props): JSX.Element {
     }
   };
 
-  const removeCollateralErrors = {
+  const addOrRemoveCollateralErrors = {
     HigherThanAvailableCollateral: "Amount entered is higher than collateral",
     BelowMinCollateralRatio: "Vault does not meet min. collateral ratio",
     ZeroRequiredTokenShare:
       "Insufficient DFI and/or DUSD in vault to maintain active loans",
-    DusdAffectVault: "Active DUSD loans require 50% DFI collaterals",
+    DusdAffectVault:
+      "Active DUSD loans require 50% DFI or 100% DUSD collaterals",
+    InsufficientBalance: "Insufficient Balance",
   };
 
+  const isDFILessThanHalfOfRequiredCollateral = tokenCollateralValue.isLessThan(
+    new BigNumber(vault.loanValue)
+      .multipliedBy(vault.loanScheme.minColRatio)
+      .dividedBy(100)
+      .dividedBy(2),
+  );
+  const getAddCollateralErrorMessage = ():
+    | AddOrRemoveCollateralError
+    | undefined => {
+    if (formState.errors.collateralAmount?.type === "hasSufficientFunds") {
+      return {
+        testID: "add_collateral_insufficient_balance",
+        message: addOrRemoveCollateralErrors.InsufficientBalance,
+      };
+    } else if (
+      hasLoan &&
+      vault.loanAmounts.some((loan) => loan.displaySymbol === "DUSD") &&
+      vault.collateralAmounts.every((col) => col.displaySymbol === "DUSD")
+    ) {
+      if (
+        !["DFI", "DUSD"].includes(selectedCollateralItem.token.displaySymbol) ||
+        (selectedCollateralItem.token.displaySymbol === "DFI" &&
+          isDFILessThanHalfOfRequiredCollateral)
```

Comment:
I guess here is a small issue, if I have a vault with DFI + DUSD as collateral and I take a small DUSD loan which is below that 50% DFI requirement. and then try to take out all DFI while staying above the collateral ratio. I would have still a valid vault because only DUSD are in as collateral.